src/dune_pkg/fetch.ml

```diff
@@ -0,0 +1,73 @@
+open Stdune
+module Process = Dune_engine.Process
+module Display = Dune_engine.Display
+
+module FiberJob = struct
+  let run command =
+    let open Fiber.O in
+    let prog =
+      command |> OpamProcess.command_cmd |> Path.Outside_build_dir.of_string
+      |> Path.outside_build_dir
+    in
+    let args = OpamProcess.command_args command in
+    let dir =
+      command |> OpamProcess.command_dir
+      |> Option.map ~f:(fun path ->
+             path |> Path.Outside_build_dir.of_string |> Path.outside_build_dir)
+    in
+    let display = Display.Quiet in
+    let env = command |> OpamProcess.command_env |> Option.map ~f:Env.of_unix in
+    let stdin_from =
+      match command |> OpamProcess.command_allow_stdin with
+      | Some true -> Some Process.Io.stdin
+      | None | Some false -> None
+    in
+    let stderr_to = Process.Io.stderr in
+    let stdout_to = Process.Io.stdout in
+    let* times =
+      Process.run_with_times ~display ?dir ?env ?stdin_from ~stderr_to
+        ~stdout_to prog args
+    in
+    (* Process.run_with_times forces Strict failure-mode, so the return code is 0 *)
+    let r_code = 0 in
+    let r_duration = times.elapsed_time in
+    let result : OpamProcess.result =
+      { r_code
+      ; r_signal = None
+      ; r_duration
+      ; r_info = []
+      ; r_stdout = []
+      ; r_stderr = []
+      ; r_cleanup = []
+      }
+    in
+    Fiber.return result
+
+  let run =
+    let open Fiber.O in
+    let rec run1 = function
+      | OpamProcess.Job.Op.Done x -> Fiber.return x
+      | Run (cmd, cont) ->
+        let* r = run cmd in
+        let k = cont r in
+        run1 k
+    in
+    run1
+end
+
+let fetch url ~target =
+  let open Fiber.O in
+  let path = Path.to_string target in
+  let fname = OpamFilename.of_string path in
+  let hashes = [] in
+  let complete = Fiber.Ivar.create () in
+  let job = OpamRepository.pull_file path fname hashes [ url ] in
```

Comment:
1. There are some calls to `OpamConsole.msg` but these are not printed in the default settings. I am not sure if this is a problem.
2. Yes, it will write to the OPAM cache. We have two options:
   1. Disable the cache altogether
   2. Specify our own cache dir, but if so, which directory to pick?

Reply:
1. ok good. that's good enough for us.
2. not sure if sharing the cache is a good idea, but for now it doesn't matter too much. i'd just leave a comment about this so that we're aware.

Reply:
In any case, `OpamConsole` is now stubbed out in our fork so it will never print anything, no matter the OPAM configuration.